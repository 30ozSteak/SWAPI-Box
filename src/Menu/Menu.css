@import "/Users/imsteaky/Turing/mod3/swapibox/src/fontawesome-free-5.3.1-web";

.menu {
  transform: rotate(0deg);
  width: 50px;
  height: 50px;
  position: fixed;
  left: 0;
  right: 0;
  margin: 0 auto;
  bottom: 5%;
  z-index: 1000;
}

.menu li {
  position: absolute;
  top: 0;
  left: 0;
  list-style: none;
  width: 100%;
  height: 100%;
  background: black;
  border: 1px solid rgba(255, 215, 0, 1);
  transition: 0.5s;
  overflow: hidden;
  transform: scale(0);
  border-radius: 50%;
  background-color: black;
}

.menu ul.active li {
  transform: scale(1);
  background-color: black;
}

.menu ul.active li:nth-child(1) {
  top: -0%;
  left: -120%;
  transition-delay: 0s;
  background: center url(../Assets/boba-fett.png);
  background-size: 30px;
  background-repeat: no-repeat;
  background-color: black;
}

.menu ul.active li:nth-child(2) {
  top: -110%;
  left: -60%;
  transition-delay: 0.2s;
  background: center url(../Assets/death-star.svg);
  background-size: 30px;
  background-repeat: no-repeat;
  background-color: black;
}

.menu ul.active li:nth-child(3) {
  top: -110%;
  left: 60%;
  transition-delay: 0.4s;
  background: center url(../Assets/millennium-falcon.svg);
  background-size: 30px;
  background-repeat: no-repeat;
  background-color: black;
}

.menu ul.active li:nth-child(4) {
  top: -0%;
  left: 120%;
  transition-delay: 0.6s;
  background: center url(../Assets/mark-as-favorite-star.svg);
  background-size: 30px;
  background-color: black;
  background-repeat: no-repeat;
}

.menu-button {
  position: fixed;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 1;
  cursor: pointer;
  transition: 0.5s;
  border: 1px solid rgba(255, 215, 0, 1);
  border-radius: 50%;
  animation: ❤️ 1.5s linear infinite;
  background: center url(../Assets/rebel-alliancewhite.svg);
  background-size: 30px;
  background-repeat: no-repeat;
  background-color: black;
}

@keyframes ❤️ {
  0% {
    box-shadow: 0 0 0 0 rgba(255, 217, 0, 0);
  }
  50% {
    box-shadow: 0 0 0 3px rgb(255, 217, 0);
  }
  100% {
    box-shadow: 0 0 0 5px rgba(255, 217, 0, 0);
  }
}

.menu-button:hover {
  background: center url(../Assets/rebel-allianceblack.svg);
  background-size: 30px;
  background-repeat: no-repeat;
  background-color: #ffd700;
}
