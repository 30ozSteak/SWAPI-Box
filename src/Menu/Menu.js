import React, { Component } from "react";
import PropTypes from "prop-types";
import MenuButtons from "../MenuButtons/MenuButtons";
import "./Menu.css";

export default class Menu extends Component {
  constructor() {
    super();
    this.state = {
      buttonState: ""
    };
  }

  toggleActive = () => {
    this.state.buttonState === ""
      ? this.setState({ buttonState: "active" })
      : this.setState({ buttonState: "" });
  };

  render() {
    return (
      <div className="menu">
        <div className="menu-button" onClick={this.toggleActive}>
          <ul className={this.state.buttonState}>
            {this.props.data.map(swRoutes => {
              return <MenuButtons key={swRoutes.swLink} data={swRoutes} />;
            })}
          </ul>
        </div>
      </div>
    );
  }
}

Menu.propTypes = {
  data: PropTypes.array
};
