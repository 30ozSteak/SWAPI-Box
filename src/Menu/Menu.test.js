import React from 'react'
import Menu from './Menu'
import { shallow } from 'enzyme'
import { mount } from 'enzyme'

describe('Menu', () => {

  it('should match the screenshot', () => {
    const mockProps = [{service: 'people'}, {service: 'planets'}]
    const wrapper = shallow(<Menu data={mockProps} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should have a default state', () => {
    const mockProps = [{service: 'people'}, {service: 'planets'}]
    const wrapper = shallow( <Menu data={mockProps} />)
    const expected = { buttonState: '' }
    expect(wrapper.state()).toEqual(expected)
  })

  it('should have a class of inactive as default', () => {
    const mockProps = [{service: 'people'}, {service: 'planets'}]
    const mockToggle = jest.fn()
    const wrapper = shallow( <Menu mockToggle={mockToggle} data={mockProps} inactive={true} />)
    expect(wrapper.is('.active')).toBe(false)
  })

  it.skip('should toggle the class on click', () => {
    const mockToggle = jest.fn()
    const mockProps = [{service: 'people'}, {service: 'planets'}]

    const wrapper = shallow(<Menu data={mockProps}
                              mockToggle={mockToggle}
                            />)
    
    wrapper.find('.menu-button').simulate('click');
    expect(mockToggle).toHaveBeenCalled();
  })

})