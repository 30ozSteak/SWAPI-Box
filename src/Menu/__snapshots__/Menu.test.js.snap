// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the screenshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    data={
      Array [
        Object {
          "service": "people",
        },
        Object {
          "service": "planets",
        },
      ]
    }
    favorites={
      Array [
        Object {
          "name": "Luke",
        },
        Object {
          "name": "Aldeern",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="menu-button"
        onClick={[Function]}
      >
        <ul
          className=""
        >
          <div>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/people"
            />
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/planets"
            />
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/vehicles"
            />
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="favorites"
            />
            <p
              className="num-faves"
            >
               
              2
               favorites saved 
            </p>
          </div>
        </ul>
      </div>,
      "className": "menu",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className=""
        >
          <div>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/people"
            />
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/planets"
            />
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/vehicles"
            />
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="favorites"
            />
            <p
              className="num-faves"
            >
               
              2
               favorites saved 
            </p>
          </div>
        </ul>,
        "className": "menu-button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/people"
            />
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/planets"
            />
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/vehicles"
            />
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="favorites"
            />
            <p
              className="num-faves"
            >
               
              2
               favorites saved 
            </p>
          </div>,
          "className": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/people"
              />,
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/planets"
              />,
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/vehicles"
              />,
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="favorites"
              />,
              <p
                className="num-faves"
              >
                 
                2
                 favorites saved 
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "onClick": undefined,
                "to": "/people",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "onClick": undefined,
                "to": "/planets",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "onClick": undefined,
                "to": "/vehicles",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "to": "favorites",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  2,
                  " favorites saved ",
                ],
                "className": "num-faves",
              },
              "ref": null,
              "rendered": Array [
                " ",
                2,
                " favorites saved ",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "ul",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="menu-button"
          onClick={[Function]}
        >
          <ul
            className=""
          >
            <div>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/people"
              />
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/planets"
              />
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/vehicles"
              />
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="favorites"
              />
              <p
                className="num-faves"
              >
                 
                2
                 favorites saved 
              </p>
            </div>
          </ul>
        </div>,
        "className": "menu",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className=""
          >
            <div>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/people"
              />
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/planets"
              />
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/vehicles"
              />
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="favorites"
              />
              <p
                className="num-faves"
              >
                 
                2
                 favorites saved 
              </p>
            </div>
          </ul>,
          "className": "menu-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/people"
              />
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/planets"
              />
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/vehicles"
              />
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="favorites"
              />
              <p
                className="num-faves"
              >
                 
                2
                 favorites saved 
              </p>
            </div>,
            "className": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/people"
                />,
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/planets"
                />,
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/vehicles"
                />,
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="favorites"
                />,
                <p
                  className="num-faves"
                >
                   
                  2
                   favorites saved 
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "onClick": undefined,
                  "to": "/people",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "onClick": undefined,
                  "to": "/planets",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "onClick": undefined,
                  "to": "/vehicles",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "to": "favorites",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    2,
                    " favorites saved ",
                  ],
                  "className": "num-faves",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  2,
                  " favorites saved ",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "ul",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
